pkg/core/state/notification_event.go

```diff
@@ -79,3 +81,42 @@ func (aer *AppExecResult) DecodeBinary(r *io.BinReader) {
 	}
 	r.ReadArray(&aer.Events)
 }
+
+// notificationEventAux is an auxiliary struct for NotificationEvent JSON marshalling.
+type notificationEventAux struct {
+	ScriptHash util.Uint160    `json:"contract"`
+	Name       string          `json:"eventname"`
+	Item       json.RawMessage `json:"state"`
+}
+
+// MarshalJSON implements implements json.Marshaler interface.
+func (ne *NotificationEvent) MarshalJSON() ([]byte, error) {
+	item, err := stackitem.ToJSONWithTypes(ne.Item)
+	if err != nil {
+		return nil, err
```

Comment:
Sometimes we (on every error (?)) stack is marshaled to the string `error: recursive reference`.
https://github.com/neo-project/neo-modules/blob/master/src/ApplicationLogs/LogReader.cs#L61